### PR TITLE
Resolves #24 - Add ability to sign in

### LINKED ISSUE #24: Add ability to sign in
### Description 

Implement logic to sign in using the Andela authentication API.

#### Developers's notes
- Use the Andela authentication API for sign in. Use `https://api.andela.com/login?redirect_url=<appurl>` in production or `https://api-staging.andela.com/login?redirect_url=<appurl>` in development and staging.
- Store the token in local storage with the key `jwt-token` to enable Single Sign-On (SSO) with other Andela Apps (AIS, Lv, Lenken, etc.)

- The production and stating authentication APIs generate tokens with different user Id's so, watch out for unexpected behavior. A quick fix would be to log out and log in again.

### PR DESCRIPTION
**Pull request by mkiterian**
(Feature #24) User should be able to sign in via the andela api

**Description**
A user should be able to sign in into the application
- When the user clicks on the sign in button,  the user is redirected to a Google Signin via the Andela API. Once the correct credentials are provided a token is set in a cookie. The token is checked for validity i.e, has not expired, has Andelan role
- If token also has Fellow role, the user is sent to the my-activities page, otherwise user is redirected to  the sign in page

**How to test**
* Add `soc-dev.andela.com` to `/etc/hosts`
- Run the application
- With a correct fellow email address, one should be able to sign in and access the my activities page
- With a non-Andelan gmail address one should see the sign in error notification

screen shot 2018-03-15 at 10 31 08

### CODE REVIEW

**Comment by john555** [comment]
Work on coverage.

File: src/components/svgIcons/notificationIcons/Error.jsx [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
```
Comment (john555) [inline src/components/svgIcons/notificationIcons/Error.jsx]: Add a snapshot test for this component.

File: tests/containers/SignIn.test.jsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import { mount } from 'enzyme';
+import { MemoryRouter } from 'react-router-dom';
+import localStorage from 'mock-local-storage';
```
Comment (john555) [inline tests/containers/SignIn.test.jsx]: `localStorage` is defined but never used.

**Comment by Farhanisto** [comment]
Looks good

File: src/assets/scss/reset.scss [hunk 3]
```diff
@@ -8,3 +8,8 @@ body {
     font-family: Roboto, sans-serif;
     margin: 0;
 }
+
+a, a:visited, a:hover, a:active {
```
Comment (john555) [inline src/assets/scss/reset.scss]: This CSS block is affecting the Breadcrumb component and will affect other links on the page.

File: package.json [hunk 4]
```diff
@@ -7,11 +7,11 @@
     "start": "webpack-dev-server --config=webpack.config.dev.js --open --history-api-fallback",
     "build": "webpack -p --config=webpack.config.prod.js && npm run copy:favicon",
     "serve": "node server.js",
-    "test": "jest --verbose",
+    "test": "export NODE_ENV=testing && jest --verbose",
     "test:dev": "jest --verbose --watch",
```
Comment (john555) [inline package.json]: Add `export NODE_ENV=testing && ` here too.

File: tests/containers/SignIn.test.jsx (hunk 2 above)
Comment (mkiterian) [reply tests/containers/SignIn.test.jsx]: This is required to mock localStorage. Without it this `window.localStorage = global.localStorage` won't work

**Pull request merged**